java/src/com/ibm/streamsx/topology/spl/SPLStreams.java

```diff
@@ -43,9 +45,19 @@
      */
     public static SPLStream subscribe(TopologyElement te, String topic,
             StreamSchema schema) {
+    	return subscribe(te, Value.of(topic), schema);
+    }
 
-        Map<String, Object> params = new HashMap<>();
+    public static SPLStream subscribe(TopologyElement te, Supplier<String> topic, StreamSchema schema) {
+    	return _subscribe(te, topic, schema);
+    }
 
+    private static SPLStream _subscribe(TopologyElement te, Supplier<String> topic, StreamSchema schema) {
```

Comment:
Not sure why an `_subscribe` was added, it's not needed the way it's coded, though it might be needed with `Object` as the type of `topic`.